### ISSUE TITLE
Core broken due to subnet_private_endpoint_network_policies_enabled

### ISSUE THREAD

**Issue by NissesSenap** [XenitAB/terraform-modules#950]
This issue was created in https://github.com/XenitAB/terraform-modules/pull/943

```shell
│ Error: Conflicting configuration arguments
│ 
│   with module.core.azurerm_subnet.this["sn-dev-we-core-servers"],
│   on .terraform/modules/core/modules/azure/core/subnets.tf line 13, in resource "azurerm_subnet" "this":
│   13:   enforce_private_link_endpoint_network_policies = local.subnet_private_endpoints[each.value.subnet_short_name]
│ 
│ "enforce_private_link_endpoint_network_policies": conflicts with private_endpoint_network_policies_enabled
```

It looks like there is a bug in the provider:

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/subnet#private_endpoint_network_policies_enabled

Even if I disable `private_endpoint_network_policies_enabled` it still complains. We could probably solve this with a dynamic resources as well.
But I will first see if there is any issues open around it.

**Comment by NissesSenap** [comment]
FYI @linus-lysell

**Comment by CalleB3** [comment]
This was fixed in #952
